contract/src/main.rs

```diff
@@ -914,54 +1011,53 @@ pub extern "C" fn set_approval_for_all() {
     )
     .unwrap_or_revert();
 
-    let operators_seed_uref = utils::get_uref(
-        OPERATOR,
-        NFTCoreError::MissingStorageUref,
-        NFTCoreError::InvalidStorageUref,
-    );
-
-    let callers_owned_tokens = match utils::get_reporting_mode() {
-        OwnerReverseLookupMode::NoLookUp => utils::get_owned_token_ids_by_token_number(),
-        OwnerReverseLookupMode::Complete => utils::get_owned_token_ids_by_page(),
-    };
+    // If caller tries to approve itself as operator that's probably a mistake and we revert.
+    if caller == operator {
+        runtime::revert(NFTCoreError::InvalidAccount);
+    }
 
     // Depending on approve_all we either approve all or disapprove all.
-    let operator = if approve_all { Some(operator) } else { None };
-    for token_id in &callers_owned_tokens {
-        // We assume a burned token cannot be approved
-        if utils::is_token_burned(token_id) {
-            continue;
-        }
+    let owner_operator_item_key = utils::key_and_value_to_str(&caller, &operator);
+    utils::upsert_dictionary_value_from_key(
+        OPERATORS,
+        &owner_operator_item_key,
+        if approve_all { Some(operator) } else { None },
```

Comment:
Hi, this dictionary stores the key (owner/operator) and Option<Key> as value indeed. I guess it could store a boolean but then we can not query the dictionary from the value (not sure it this makes sense anyway) and I thought this would make more sense to be able to have the direct operator as value here. Let me try with a boolean though as I guess reverse lookup from dictionary value is anyway not possible.

Reply:
Please check that commit 

https://github.com/casper-ecosystem/cep-78-enhanced-nft/pull/181/commits/7314d36cb43369fbe178eaa3bca1a84c70ed04b8

I am a bit circumspect of the difference between APPROVED and OPERATORS. now, one deal with Option::<Key>, the second with booleans for values which is not super consistent but it works as well. It's probably lighter on storage to save a boolean than an account though.